shared-worker/firebase-worker.js

```diff
@@ -0,0 +1,54 @@
+importScripts('../firebase/firebase-app.js');
+importScripts('../firebase/firebase-firestore.js');
+var config = {
+  // your config here
+};
+
+/** Port used to communicate with the shared worker. */
+self.sharedWorkerPort = null;
+
+/** Whether this Firestore worker holds the lease. */
+self.hasPrimaryLease = false;
+
+/** The Firestore instance. */
+self.db = null;
+
+self.onmessage = function(e) {
+  // Setup message listeners after receiving the Shared Worker's port and
+  // request the lease.
+  if (e.data.sharedWorkerPort) {
+    self.sharedWorkerPort = e.data.sharedWorkerPort;
+    self.sharedWorkerPort.onmessage = (event) => handleLeaseEvent(event);
+    self.sharedWorkerPort.postMessage('requestingLease');
+  } 
+
+  // Release the primary lease in the shared worker if this worker owns the
+  // lease when this worker is being terminated.
+  if (e.data === 'terminate' && self.hasPrimaryLease) {
+    self.sharedWorkerPort.postMessage('releaseLease');
+  } 
+};
+
+async function initializeApp(takeLease) {
+  const app2 = firebase.initializeApp(config, "some-namez");
```

Comment:
why "app2" and "some-namez" instead of the default app?

Reply:
you caught me 😨